nomad/encrypter.go

```diff
@@ -317,14 +301,33 @@ func (e *Encrypter) RemoveKey(keyID string) error {
 
 // saveKeyToStore serializes a root key to the on-disk keystore.
 func (e *Encrypter) saveKeyToStore(rootKey *structs.RootKey) error {
-	var buf bytes.Buffer
-	enc := codec.NewEncoder(&buf, structs.JsonHandleWithExtensions)
-	err := enc.Encode(rootKey)
+
+	kek, err := helper.CryptoBytes(32)
+	if err != nil {
+		return fmt.Errorf("failed to generate key wrapper key: %v", err)
+	}
+	wrapper, err := e.newKMSWrapper(rootKey.Meta.KeyID, kek)
+	if err != nil {
+		return err
```

Comment:
In this case this is arguably a programmer error rather than a runtime error, so I'd left it off. But if we add more KMS implementations that won't be the case, so yeah better to wrap it.